### DISCUSSION TITLE
Solar radiation from Solar Panels?

### DISCUSSION THREAD

**Discussion by Taraman17** [jeroenterheerdt/HAsmartirrigation#367] [General]
Hi,
since I don't have any sensor for measuring solar radiation, I was just wondering if there might be a way to derive that from my PV-data?

**Comment by jeroenterheerdt** [comment]
sorry, what is PV?

**Comment by cryptedx** [comment]
He is meaning Photovoltaik = Solar

**Comment by jeroenterheerdt** [comment]
ah thanks - I am not sure how that would work, I guess you can somehow take the values from your PV system and convert them to solar radiation, but I don't know how and it's probably dependent on your PV system.

**Comment by cryptedx** [comment]
Thank you for your great question. I also ask that to me some days ago but did not had the motivation to figure it out by myself. Now I tried it with ChatGPT and it brought me to this solution:

```yaml
sensor:
  - platform: template
    sensors:
      garage_solar_irradiance:
        friendly_name: "Garage Solar Irradiance"
        unit_of_measurement: "W/m²"
        value_template: >
          {% set pv_power_output = states('sensor.sunny_tripower_x_25_pv_power_c') | float %}
          {% set panel_area = 25.5 %} 
          {% set panel_efficiency = 0.22 %} 
          {% if panel_area > 0 and panel_efficiency > 0 %}
            {{ (pv_power_output / (panel_area * panel_efficiency)) | round(2) }}
          {% else %}
            0
          {% endif %}
```
I have set it up but it's dark now. Let's see how it will perform tomorrow.

**Reply by Taraman17** [reply]
That sounds like a good starting point.
I will experiment with that the next days.

**Reply by cryptedx** [reply]
It looks like to be working for me. Now let's see what irrigation values I will get in the next days.

**Reply by Taraman17** [reply]
So, the sensor is working.
I wonder if that if-statement in the template is really needed, since the variables are set statically and will never be 0, so the if statement will always be true. 

I will see, what irrigation values I get from this and then tweak it to my needs.
Since I have a very cluttered irrigation system, in which I have to estimate the values anyway, I will change it so, the values make sense to me and use these.

**Comment by mVik85** [comment]
This was bothering me, as solar data is available just like weather.
I ended up with an API call to Forecast.Solar every 30 min (free), it gets me watts with hourly resolution, and from this raw data Gemini helped me to create a sensor with interpolation which is updated every minute.

Would anyone be interested?

And I'm just wondering what estimation might PyETO do under the hood to compare it. If useful it could be included easily into the integration.
